### PR TITLE
Ryna - LB171.LC_R_Cropland_Yh_GLU_irr

### PR DESCRIPTION
**Pull request by rynacui**
#377

### CODE REVIEW

File: R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
 #' @importFrom dplyr filter mutate select
 #' @importFrom tidyr gather spread
-#' @author YourInitials CurrentMonthName 2017
-#' @export
-module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr_DISABLED <- function(command, ...) {
+#' @author RC May 2017
+module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr <- function(command, ...) {
   if(command == driver.DECLARE_INPUTS) {
     return(c( "L122.LC_bm2_R_HarvCropLand_C_Yh_GLU",
-              "L161.ag_irrHA_bm2_R_C_Y_GLU",
```
Comment (rynacui) [inline R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R]: These two files are not used, removed from precursors.

File: R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R [hunk 2]
```diff
@@ -33,76 +30,139 @@ module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr_DISABLED <- function(command, ...) {
...
+      # Match the base year share by GCAM region, commodity, year and GLU.
+      left_join_error_no_match(L171.ag_irrHA_frac_R_C_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      # Calculate the rainfed cropland cover in all historical periods.
+      mutate(value = total * rfd_share,
+             # For the rainfed cropland table, missing values default to total cropland quantities; there are no missing currently (5/10/17).
+             value = if_else(is.na(value), total, value)) %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU, year, value) ->
+      L171.LC_bm2_R_rfdHarvCropLand_C_Yh_GLU
+
+    # Extend irrigated cropland cover data to all years, 1700-2010
```
Comment (rynacui) [inline R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R]: Here, interpolating the data back to 1700 is only to pass the `timeshift` test, but those years are not used for calculation. There is in fact no missing data for the historical years 1971-2010. I've also tried just interpolating between 1971-2010 to double check, and it failed the `timeshift` test on `left_join_error_no_match`.

File: R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R (hunk 1 above)
Comment (bpbond) [inline R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R]: Nice catch-thanks.

File: R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R [hunk 3]
```diff
@@ -33,76 +30,139 @@ module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr_DISABLED <- function(command, ...) {
...
+    id_list %>%
+      repeat_add_columns(all_years) %>%
+      left_join(L171.LC_bm2_R_irrHarvCropLand_C_Yh_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "year")) %>%
+      group_by(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      mutate(value = approx_fun(year, value)) %>%
+      ungroup() -> irrHarvCropLand_interp
+
+    # Extend rainfed and cropland cover data to all years, 1700-2010
+    # Note: 1700, 1750, 1800, 1850, 1900, and 1950 are all zero; and this interpolation fills 1700-1970 with zeros, but no missing elsewhere.
+    id_list <- L171.LC_bm2_R_rfdHarvCropLand_C_Yh_GLU %>%
```
Comment (bpbond) [inline R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R]: Be consistent - all your other pipelines end with assignment (`->`) so do so here too?

**Pull request merged**